modules/andes-core/broker/src/main/java/org/wso2/andes/server/information/management/QueueManagementInformationMBean.java

```diff
@@ -322,7 +322,7 @@ public void restoreMessagesFromDeadLetterQueue(@MBeanOperationParameter(name = "
             description = "The Dead Letter Queue Name for the selected tenant") String destinationQueueName) {
 
         if (null != andesMetadataIDs) {
-            List<Long> andesMessageIdList = Arrays.asList(ArrayUtils.toObject(andesMetadataIDs));
+            List<Long> andesMessageIdList = new ArrayList<>(Arrays.asList(ArrayUtils.toObject(andesMetadataIDs)));
```

Comment:
This doesn't hit the performance critical path. This part of the code is only run when a user browses the dead letter channel and restores/re-routes messages from it. Therefore, I did add all. Fixed in the PR https://github.com/wso2/andes/pull/381